mpmath/matrices/linalg.py

```diff
@@ -787,3 +839,58 @@ def qr(ctx, A, mode = 'full', edps = 10):
         # ------------------
         # END OF FUNCTION QR
         # ------------------
+
+    def rank(ctx, A, iszerofunc=None):
+        """
+            Calculate the rank of a matrix,
+            i.e. the number of linear independent
+            columns (or rows equivalently).
+
+            Rank is computed via singular value decomposition
+            by counting the number of non-zero singular values.
```

Comment:
Could you please document the `iszerofunc` parameter, and maybe add an example using it?

Reply:
added tests some example for `iszerofunc`